rskj-core/src/main/java/co/rsk/net/BlockSyncService.java

```diff
@@ -102,18 +102,28 @@ public BlockProcessResult processBlock(@Nonnull Block block, Peer sender, boolea
                 logger.trace("Missing hashes for block in process {} {}", blockNumber, block.getPrintableHash());
                 requestMissingHashes(sender, unknownHashes);
             }
-            return BlockProcessResult.invalidBlock(block, start);
+            return false;
+        }
+
+        return true;
+    }
+
+    public BlockProcessResult processBlock(@Nonnull Block block, Peer sender, boolean ignoreMissingHashes) {
+        final Instant start = Instant.now();
+
+        boolean readyForProcessing = preprocessBlock(block, sender, ignoreMissingHashes);
```

Comment:
We could add the relay headers validations here, so we prevent accessing the disk before PoW check

Reply:
Done in 2db0d6e6f1d60a415b0456338f54c34e673879f8